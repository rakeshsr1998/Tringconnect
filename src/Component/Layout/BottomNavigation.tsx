import React, {useState} from 'react';
import {Image, Pressable, View} from 'react-native';
import {style} from './bottom-style';

export const BottomNavigation = () => {
  const [index, setIndex] = useState(0);
  return (
    <View style={style.container}>
      <View style={style.subContainer}>
        <Pressable onPress={() => setIndex(0)}>
          <Image
            source={require('../../assets/home-filled.png')}
            style={style.menu}
          />
          {index === 0 && <View style={style.bar} />}
        </Pressable>
        <Pressable onPress={() => setIndex(1)}>
          <Image
            source={require('../../assets/dull-calender.png')}
            style={style.menu}
          />
          {index === 1 && <View style={style.bar} />}
        </Pressable>
        <Pressable onPress={() => setIndex(2)}>
          <Image
            source={require('../../assets/dull-post.png')}
            style={style.menu}
          />
          {index === 2 && <View style={style.bar} />}
        </Pressable>
        <Pressable onPress={() => setIndex(3)}>
          <Image
            source={require('../../assets/dull-folder.png')}
            style={style.menu}
          />
          {index === 3 && <View style={style.bar} />}
        </Pressable>
        <Pressable onPress={() => setIndex(4)}>
          <Image
            source={require('../../assets/dull-chat.png')}
            style={style.menu}
          />
          {index === 4 && <View style={style.bar} />}
        </Pressable>
      </View>
    </View>
  );
};
