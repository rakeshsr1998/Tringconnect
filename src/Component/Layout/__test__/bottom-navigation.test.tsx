import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { BottomNavigation } from '../BottomNavigation';

describe('<Header>', () => {
  const mockFunction = jest.fn();
  beforeEach(() => {
    const component = <BottomNavigation />;
    instance = render(component);
  });

  /**
   * @type {RenderAPI}
   */
  let instance: any;

  it('Should render header with home click', async () => {
    const nextButton = instance.getByTestId('home');
    fireEvent(nextButton, 'onPress');

    const chat = instance.getByTestId('chat');
    fireEvent(chat, 'onPress');

    const folder = instance.getByTestId('folder');
    fireEvent(folder, 'onPress');

    const cal = instance.getByTestId('cal');
    fireEvent(cal, 'onPress');

    const post = instance.getByTestId('post');
    fireEvent(post, 'onPress');

  });
});