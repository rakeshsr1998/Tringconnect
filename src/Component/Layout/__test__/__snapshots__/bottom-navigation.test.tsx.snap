// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> Should render header with call click 1`] = `[MockFunction]`;

exports[`<Header> Should render header with chat click 1`] = `[MockFunction]`;

exports[`<Header> Should render header with folder click 1`] = `[MockFunction]`;

exports[`<Header> Should render header with home click 1`] = `[MockFunction]`;

exports[`<Header> Should render header with home click 2`] = `[MockFunction]`;

exports[`<Header> Should render header with home click 3`] = `[MockFunction]`;

exports[`<Header> Should render header with home click 4`] = `[MockFunction]`;

exports[`<Header> Should render header with home click 5`] = `[MockFunction]`;

exports[`<Header> Should render header with post click 1`] = `[MockFunction]`;
